clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -559,8 +545,6 @@ private long endTimestamp() {
         } catch (KafkaException e) {
             if (fetched.isEmpty())
                 throw e;
-            // To be thrown in the next call of this method
-            nextInLineExceptionMetadata = new ExceptionMetadata(fetchedPartition, fetchedOffset, e);
```

Comment:
Good idea to consolidate the error handling (I was also wondering if it was possible). Since we're catching the exception in `PartitionRecords.fetchRecords`, do we still need this try/catch?